diff --git a/Sources/CSSOM/Generated.swift b/Sources/CSSOM/Generated.swift
index 16a452b..1242f02 100644
--- a/Sources/CSSOM/Generated.swift
+++ b/Sources/CSSOM/Generated.swift
@@ -378,7 +378,8 @@ public class CSSColorValue: CSSStyleValue {
         super.init(unsafelyWrapping: jsObject)
     }
 
-    @inlinable override public class func parse(cssText: String) -> CSSColorValue_or_CSSStyleValue {
+    // returns CSSStyleValue | CSSColorValue
+    @inlinable public class func parse(cssText: String) -> CSSStyleValue {
         let this = constructor!
         return this[Strings.parse].function!(this: this, arguments: [_toJSValue(cssText)]).fromJSValue()!
     }
@@ -935,7 +936,7 @@ public class CSSNumericValue: CSSStyleValue {
         return this[Strings.type].function!(this: this, arguments: []).fromJSValue()!
     }
 
-    @inlinable override public class func parse(cssText: String) -> Self {
+    @inlinable public class func parse(cssText: String) -> Self {
         let this = constructor!
         return this[Strings.parse].function!(this: this, arguments: [_toJSValue(cssText)]).fromJSValue()!
     }
