diff --git a/Sources/WebAudio/Generated.swift b/Sources/WebAudio/Generated.swift
index 35bbf34..770a8e8 100644
--- a/Sources/WebAudio/Generated.swift
+++ b/Sources/WebAudio/Generated.swift
@@ -194,7 +194,8 @@ public class AudioBufferSourceNode: AudioScheduledSourceNode {
     @ReadWriteAttribute
     public var loopEnd: Double

-    @inlinable override public func start(when: Double? = nil, offset: Double? = nil, duration: Double? = nil) {
+    // `override` removed since the superclass function has fewer parameters
+    @inlinable func start(when: Double? = nil, offset: Double? = nil, duration: Double? = nil) {
         let this = jsObject
         _ = this[Strings.start].function!(this: this, arguments: [_toJSValue(when), _toJSValue(offset), _toJSValue(duration)])
     }
